File: lesson_2/app/src/main/java/com/kalashnyk/denys/retrofitexample/presentation/item/UserViewHolder.java
package com.kalashnyk.denys.retrofitexample.presentation.item;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.kalashnyk.denys.retrofitexample.R;
import com.kalashnyk.denys.retrofitexample.pojo.User;

public class UserViewHolder extends RecyclerView.ViewHolder {

    private View mItemView;
    private User mUser;

    public UserViewHolder(@NonNull View itemView) {
        super(itemView);
        mItemView = itemView;
    }

    public void bind(User user) {
        mUser = user;
        setupItem();
    }

    private void setupItem() {
        TextView tvId = mItemView.findViewById(R.id.txt_rv_id);
        TextView tvName = mItemView.findViewById(R.id.txt_name);
        TextView tvSurName = mItemView.findViewById(R.id.txt_surname);
        TextView tvFatherName = mItemView.findViewById(R.id.txt_fathername);

        tvId.setText(String.valueOf(mUser.getId()));
        tvName.setText(mUser.getName());
        tvSurName.setText(mUser.getSurname());
        tvFatherName.setText(mUser.getFathername());
    }
}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/presentation/item/UserViewHolder.kt
package com.kalashnyk.denys.kotlinsample.presentation.item

import android.support.v7.widget.RecyclerView
import android.view.View
import com.kalashnyk.denys.kotlinsample.repository.database.entity.UserEntity
import kotlinx.android.synthetic.main.item_user.view.*

class UserViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {

    private var userEntity: UserEntity? = null
    private var listener: IUserItemClickListener<UserEntity>? = null
    private val itemDetail = View.OnClickListener { listener!!.openDetail(this!!.userEntity!!) }

    fun bind(user: UserEntity, listener: IUserItemClickListener<UserEntity>) {
        userEntity = user
        this.listener = listener
        setupItem()
    }

    private fun setupItem() {
        view.txtRvId.text = userEntity?.id.toString()
        view.txtName.text = userEntity?.name
        view.txtSurname.text = userEntity?.surname
        view.txtFathername.text = userEntity?.fathername
        view.setOnClickListener(itemDetail)
    }
}
